packages/auth/src/react.tsx

```diff
@@ -0,0 +1,533 @@
+      clientLogger[level] = (code: string, metadata: Error) => {
+        _logger[level](code, metadata); // Logs to console
+
+        if (level === 'error') {
+          metadata = formatError(metadata) as Error;
+        }
+        (metadata as any).client = true;
+        const url = `${basePath}/_log`;
+        const body = new URLSearchParams({ level, code, ...(metadata as any) });
+        if (navigator.sendBeacon) {
+          return navigator.sendBeacon(url, body);
+        }
+        return fetch(url, { method: 'POST', body, keepalive: true });
+      };
+    }
+    return clientLogger as unknown as LoggerInstance;
+  } catch {
+    return _logger;
+  }
+}
+
+function parseUrl(url?: string): InternalUrl {
+  const defaultUrl = new URL('http://localhost:3000/api/auth');
+
+  if (url && !url.startsWith('http')) {
+    url = `https://${url}`;
+  }
+
+  const _url = new URL(url ?? defaultUrl);
+  const path = (_url.pathname === '/' ? defaultUrl.pathname : _url.pathname)
+    // Remove trailing slash
+    .replace(/\/$/, '');
+
+  const base = `${_url.origin}${path}`;
+
+  return {
+    origin: _url.origin,
+    host: _url.host,
+    path,
+    base,
+    toString: () => base,
+  };
+}
+
+// This behaviour mirrors the default behaviour for getting the site name that
+// happens server side in server/index.js
+// 1. An empty value is legitimate when the code is being invoked client side as
+//    relative URLs are valid in that context and so defaults to empty.
+// 2. When invoked server side the value is picked up from an environment
+//    variable and defaults to 'http://localhost:3000'.
+const __NEXTAUTH = {
+  baseUrl: parseUrl(process.env.NEXTAUTH_URL ?? process.env.VERCEL_URL).origin,
+  basePath: parseUrl(process.env.NEXTAUTH_URL).path,
+  baseUrlServer: parseUrl(
+    process.env.NEXTAUTH_URL_INTERNAL ??
+      process.env.NEXTAUTH_URL ??
+      process.env.VERCEL_URL
+  ).origin,
+  basePathServer: parseUrl(
+    process.env.NEXTAUTH_URL_INTERNAL ?? process.env.NEXTAUTH_URL
+  ).path,
+  _lastSync: 0,
+  _session: undefined,
+  _getSession: () => {},
+} as AuthClientConfig & {
+  _session: CloudSession | null | undefined;
+};
+
+const broadcast = BroadcastChannel();
+
+const logger = proxyLogger(_logger, __NEXTAUTH.basePath);
+
+interface CloudSession extends Session {
+  user: {
+    id: string;
+    hasPassword: boolean;
+  } & DefaultSession['user'];
+}
+
+export type SessionContextValue<R extends boolean = false> = R extends true
+  ?
+      | {
+          data: CloudSession;
+          status: 'authenticated';
+        }
+      | {
+          data: null;
+          status: 'loading';
+        }
+  :
+      | {
+          data: CloudSession;
+          status: 'authenticated';
+        }
+      | {
+          data: null;
+          status: 'unauthenticated' | 'loading';
+        };
+
+const sessionLoadingAtom = atom<boolean>(false);
+const sessionValueAtom = atom<CloudSession | null>(null);
+
+export const sessionAtom = atom<
```

Comment:
I suggest do not use jotai in this package so that it will be easier to be commited to upstream.